src/app/inbound/mod.rs

```diff
@@ -45,6 +50,9 @@ where
     let default_fwd_addr = config.inbound_forward.map(|a| a.into());
     let dispatch_timeout = config.inbound_dispatch_timeout;
 
+    let mut trace_labels = HashMap::new();
+    trace_labels.insert("direction".to_string(), "inbound".to_string());
```

Comment:
This feels weird to me; and, as discussed IRL, it feels weird to push this into the converter, specifically. I'd _expect_ all of the labels to be derived from the `Target` in the `trace_context` layer, but it sounds like some more work may be necessary to make that a reality.